blog/content/post/2020-03-27-trenchboot-nlnet-lz.md

```diff
@@ -0,0 +1,419 @@
+each pakcage is `stdenv` library, which
+[documentation](https://nixos.org/nixos/nix-pills/fundamentals-of-stdenv.html)
+should clarify some issues. However, it is good to experiment with package
+manager on your own. Good point to start are already existing packages and their
+`default.nix` files.
+
+You may wonder why do we build our own tools, when most of them are available in
+default `nixpkgs`. Answer is simple - we have possibility to build not only
+tools, but also many other software and firmware components... You will
+learn more about it in next section.
+
+## Firmware
+
+As I mentioned in previous section, with NixOS we have possibility to customize
+our platform (both firmware and operating system). Before, we do that, I need to
+introduce some concepts. The better you understand them, the more awareness
+about security of your platform you will have and better understand the purpose
+of entire project. Let's break our entire system into following main parts. They
+are listed in order of execution when the platform is booted, so you can also
+see the flow and relations between them.
+
+#### BIOS
+
+Our solution is fully open-source, so we also use such firmware (BIOS) -
+**coreboot**. Also, as we are maintainers of coreboot for PC Engines platforms,
+we use
+[this](https://github.com/pcengines/coreboot/tree/pcengines_trenchboot_4.11.x)
+particular fork and branch. Every change appears there, so it is definitely
+place where you will find up-to-date firmware. I won't describe coreboot itself
+here. What you need to know, it is first code which is running during boot
+process. It initialize all hardware components on platform.
+
+#### GRUB
+
+When coreboot ends its work, a bootloader comes in. We use **GRUB**. Its task is
+to boot operating system. Normally, if NixOS is installed on hard disk, GRUB
+loads it to RAM. Being more precise, *Linux kernel* and *initrd* are being
+loaded. However, that operation is slightly different when we enable DRTM. As it
+is main goal of our project, let's take a look a little bit closer on it.
+
+We prepared extension module called `slaunch`. When DRTM is enabled, `slaunch`
+executes AMD's dedicated machine instruction `SKINIT`. Treat is 'pre-DRTM'
+operation. It prepares all components and jump to secure loader - `Landing
+Zone`.
+
+#### Landing Zone (LZ)
+
+LZ has 2 main tasks: measure Linux kernel and then run it. If you wonder, when
+measurements are done, then you have just find out. Without LZ, DRTM doesn't
+exist. Although it seems inconspicuous, it does fundamental job in secure boot
+process.
+
+#### Linux kernel
+
+Linux kernel is a core of all Linux based operating systems. Actually, it is
+operating system without any additional applications. It can be modified to
+given platform by including and excluding particular drivers, called modules. In
+general, it is done to limit size of kernel image and adjust its performance.
```

Comment:
Linux kernel also does some measurements in the Trenchboot Secure Launch process.

Reply:
I don't mention it here, but later in 'linux-trenchboot' package description.